src/main/java/uk/gov/pay/card/db/RangeSetCardInformationStore.java

```diff
@@ -47,15 +47,12 @@ public void put(CardInformation cardInformation) {
     @Override
     public Optional<CardInformation> find(String prefix) {
         try {
-            Long.valueOf(prefix);
+            return Optional.ofNullable(rangeSet.rangeContaining(Long.valueOf(prefix)))
```

Comment:
I'd be tempted to pull out the parsing of the String and error handling to a private method to simplify it a little (one method to do one thing etc) 
```
@Override
    public Optional<CardInformation> find(String prefix) {
        return parseToLong(prefix)
                .map(rangeSet::rangeContaining)
                .map(store::get);
    }
    
    private Optional<Long> parseToLong(String value) {
        try {
            return Optional.of(Long.valueOf(value));
        } catch (NumberFormatException e) {
            LOGGER.error("Received card number that cannot be parsed into long");
            return Optional.empty();
        }
    }
```